backend/app/services/adapters/executor_job.py

```
@@ -227,6 +383,92 @@ async def cleanup_stale_executors(self, db: AsyncSession) -> None:
+        if self._is_device_executor_name(executor_name):
+            return {"reason": "device_executor"}
+
+        latest_update: datetime | None = None
+        for subtask in subtasks:
+            task = task_map.get(subtask.task_id)
+            if not task:
+                return {"reason": "task_not_found"}
+
+            try:
+                task_crd = Task.model_validate(task.json)
+            except Exception as exc:
```

Ruff (0.15.13)

[warning] Do not catch blind exception: `Exception`

(BLE001)